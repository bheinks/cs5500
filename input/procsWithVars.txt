(*
	Declarations of procedures, some nested, with
      variable declarations
*)

program p;
var bojack : integer;

procedure p1;
var mrPB : integer;

  procedure p1a;
  var todd : boolean;
  begin
    write(0)
  end; (* p1a *)

  procedure p1b;
  var princessCarolyn : char;

    procedure p1b_a;
    var bojack : integer;
    begin
      write(1)
    end; (* p1b_a *)

  begin (* p1b *)
    write(2)
  end; (* p1b *)

begin (* p1 *)
  write(3)
end; (* p1 *)

procedure p2;
var diane : integer;
begin
  write(99)
end;

begin (* main *)
  write(5)
end.

