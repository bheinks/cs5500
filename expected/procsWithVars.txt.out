

>>> Entering new scope...

+++ Adding p to symbol table with type PROGRAM

+++ Adding bojack to symbol table with type INTEGER

+++ Adding p1 to symbol table with type PROCEDURE


>>> Entering new scope...

+++ Adding mrPB to symbol table with type INTEGER

+++ Adding p1a to symbol table with type PROCEDURE


>>> Entering new scope...

+++ Adding todd to symbol table with type BOOLEAN

<<< Exiting scope...

+++ Adding p1b to symbol table with type PROCEDURE


>>> Entering new scope...

+++ Adding princessCarolyn to symbol table with type CHAR

+++ Adding p1b_a to symbol table with type PROCEDURE


>>> Entering new scope...

+++ Adding bojack to symbol table with type INTEGER

<<< Exiting scope...

<<< Exiting scope...

<<< Exiting scope...

+++ Adding p2 to symbol table with type PROCEDURE


>>> Entering new scope...

+++ Adding diane to symbol table with type INTEGER

<<< Exiting scope...

<<< Exiting scope...

---- Completed parsing ----
